parastell/radial_distance_utils.py

```diff
@@ -0,0 +1,170 @@
+    """
+    Reorder the filaments so they start near the outboard xy plane crossing,
+    and begin by increasing z value.
+
+    Arguments:
+        filaments (list of list of list of float): List of filaments, which are
+            lists of points defining each filament.
+    Returns:
+        filaments (list of list of list of float): Reordered list of filaments,
+            suitable for building the magnet surface.
+    """
+    for filament_index, filament in enumerate(filaments):
+        # start the filament at the outboard
+        max_z_index = get_start_index(filament)
+        reordered_filament = np.concatenate(
+            [filament[max_z_index:], filament[0:max_z_index]]
+        )
+
+        # make sure z is increasing initially
+        if filament[max_z_index, 2] > filament[max_z_index + 1, 2]:
+            reordered_filament = np.flip(reordered_filament, axis=0)
+
+        # remove duplicate point since the start point might be different
+        _, idx = np.unique(reordered_filament, return_index=True, axis=0)
+        reordered_filament = reordered_filament[np.sort(idx)]
+
+        # ensure filament is a closed loop
+        reordered_filament = np.concatenate(
+            [reordered_filament, [reordered_filament[0]]]
+        )
+
+        filaments[filament_index] = reordered_filament
+
+    filaments = sort_filaments_toroidally(filaments)
+
+    return filaments
+
+
+def get_reordered_filaments(magnet_set):
+    """
+    Convenience function to get the reordered filament data from a magnet
+    """
+    magnet_set._extract_filaments()
+    magnet_set._set_average_radial_distance()
+    magnet_set._set_filtered_filaments()
+
+    filtered_filaments = magnet_set.filtered_filaments
+    filaments = reorder_filaments(filtered_filaments)
+
+    return filaments
+
+
+def build_magnet_surface(reordered_filaments):
+    for fil_index, _ in enumerate(reordered_filaments[0:-1]):
+        fil1 = reordered_filaments[fil_index]
+        fil2 = reordered_filaments[fil_index + 1]
+        for index, _ in enumerate(fil1):
+            x1 = fil1[index, 0]
+            x2 = fil2[index, 0]
+            y1 = fil1[index, 1]
+            y2 = fil2[index, 1]
+            z1 = fil1[index, 2]
+            z2 = fil2[index, 2]
+            cubit.cmd(
+                f"create curve location {x1} {y1} {z1} location {x2} {y2} {z2}"
+            )
+
+    lines = np.array(cubit.get_entities("curve"))
+    lines = np.reshape(
+        lines, (len(reordered_filaments) - 1, len(reordered_filaments[0]))
+    )
+    for loop in lines:
+        for line in loop[0:-1]:
+            cubit.cmd(f"create surface skin curve {line} {line + 1}")
+
+
+def measure_radial_distance(ribs):
+    distances = []
+    for rib in ribs:
+        distance_subset = []
+        for point, direction in zip(rib.rib_loci, rib._normals()):
+            cubit.cmd(f"create vertex {point[0]} {point[1]} {point[2]}")
+            vertex_id = max(cubit.get_entities("vertex"))
```

Comment:
Isn't there a better way to get the last vertex than getting all of the vertices and taking the max?

Reply:
I think you might be able to use ```vtx_id = cubit.get_last_id("vertex")```